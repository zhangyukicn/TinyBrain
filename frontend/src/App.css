.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.Picture {
  display: flex;
  flex-direction: column;
  background-color: aqua;
  height: 350px;
  width: 50%;
  text-align: center;
  justify-content: center;
}
.picture-part {
  background-color: black;
  height: 300px;
  width: 300px;
  margin-right: auto;
  margin-left: auto;
}
.Answer{
  display: block;
  background-color: yellow;
  height: 350px;
  width:70%;
  overflow-x: scroll;
  
}
.title-part{
  background-color:white;
  height: 100px;
  width:40%;
  margin-left: auto;
  margin-right: auto;
  /* margin-bottom: 20px; */
  text-align: center;
  margin-top: 20px;
}
.words-part{
  background-color:white;
  height: 100%;
  width:50%;
  margin-left: auto;
  margin-right: auto;
  /* margin-bottom: 20px; */
  text-align: center;
  margin-top: 20px;
}
.pic-part{
  /* background-color:red; */
  height: 50px;
  width:40%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  text-align: center;
  margin-top: 20px;
  font-size: 35px;
}



.pressbutton{
  width:50%;
  height:50px;
  margin-left: 25%;
  margin-right: auto;
  margin-bottom: 10px;
  
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
