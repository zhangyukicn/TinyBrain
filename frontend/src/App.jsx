import React from 'react';
import './App.css';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
    Link
} from 'react-router-dom';
import TokenContext from './TokenContext';
import Welcome from './pages/welcome';
import Login from './pages/login';
import Signup from './pages/signup';
import Dashboard from './pages/dashboard';
import Library from './pages/library';
import PlayerDashboard from './pages/playerdashboard';
import Lobby from './pages/lobby';

// yarn add @material-ui/icons
// yarn add @material-ui/core
// yarn add react-router-dom

// 页面4： dashboard
// 页面5： library
// 页面6： view game
// 页面7： edit game
// 页面8： play game
// 页面9： game result

// 页面10： join game
// 页面11： myresult

/* <TokenContext.Provider value={{ token, setToken }}>
            <Router>
            {token ? null : <Redirect to='./login'/>}
            <Switch>
                <Route path="./login" component={Login} setToken={setToken} token={token} />
                <Route path="./signup" component={Signup} setToken={setToken} token={token} />
                <Route path="./dashboard" component={Dashboard} setToken={setToken} token={token} />
            </Switch>
            </Router>
        </TokenContext.Provider> */

function App () {
    const tokenStored = localStorage.getItem('token');
    const [token, setToken] = React.useState(tokenStored !== null ? tokenStored : '');
    // useState这个函数接收的参数是状态的初始值(Initial state)
    // 它返回一个数组，这个数组的第0位是当前的状态值，第1位是可以改变状态值的方法函数。
    React.useEffect(() => {
        localStorage.setItem('token', token);
    }, [token]); // change when token changed
    console.log(token);
    return (
        <Router>
            {/* A <Switch> looks through its children <Route>s and
                renders the first one that matches the current URL. */}
            <TokenContext.Provider value={{ token, setToken }}>
                {/* 检查有否token，没有就进欢迎页面 */}
                {token ? null : <Redirect to='/'/>}
                <Switch>
                {/* 必须用exact path 否则匹配到/就不往下匹配了 */}
                <Route exact path="/" component={Welcome} />
                <Route exact path="/login" component={Login} />
                <Route exact path="/signup" component={Signup} />
                <Route exact path="/dashboard" component={Dashboard} />
                <Route exact path="/library" component={Library} />
                <Route exact path="/playerdashboard" component={PlayerDashboard} />
                {/* <Route exact path="/lobby" component={Lobby} /> */}
                </Switch>
            </TokenContext.Provider>
        </Router>
    );
}

export default App;
