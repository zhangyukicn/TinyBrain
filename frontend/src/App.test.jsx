import React from 'react';
import App from './App';
import { render, screen } from '@testing-library/react';
import { configure, EnzymeAdapter, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TokenContext from './TokenContext';
import Welcome from './pages/welcome';
import Login from './pages/login';
import Signup from './pages/signup';
import Dashboard from './pages/dashboard';
import PlayerDashboard from './pages/playerdashboard';
import Lobby from './pages/lobby';
import Edit from './pages/editQuiz';
import EditQuestion from './pages/editQuestion';
import Playcontrol from './pages/playControl';
import PlayResult from './pages/playResult';
import PlayerplayingDashborad from './pages/playerplayingDashborad';
import Playerresult from './pages/playerresult';
import { ExpansionPanelActions } from '@material-ui/core';
configure({ adapter: new Adapter() });

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/Login/i);
//   expect(linkElement).toBeInTheDocument();
// });

// playerplayingdashbord

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: './lobby',
    state: { id: 173450416 }
  }),
}));

describe('AnswerDispay', () => {
  const modalStarta = shallow(<PlayerplayingDashborad />);

  it('is matched with snapshot', () => {
    expect(modalStarta).toMatchSnapshot();
  });
})
