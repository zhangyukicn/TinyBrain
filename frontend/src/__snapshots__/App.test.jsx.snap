// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerDispay is matched with snapshot 1`] = `ShallowWrapper {}`;


